flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/TableStoreManagedFactory.java

```diff
@@ -183,6 +186,11 @@ public void onDropTable(Context context, boolean ignoreIfNotExists) {
     @Override
     public Map<String, String> onCompactTable(
             Context context, CatalogPartitionSpec catalogPartitionSpec) {
-        throw new UnsupportedOperationException("Not implement yet");
+        Map<String, String> newOptions = new HashMap<>(context.getCatalogTable().getOptions());
+        newOptions.put(COMPACTION_MANUAL_TRIGGERED.key(), String.valueOf(true));
+        newOptions.put(
+                COMPACTION_PARTITION_SPEC.key(),
+                JsonSerdeUtil.toJson(catalogPartitionSpec.getPartitionSpec()));
```

Comment:
> Compact without partition is not supported now?

For partitioned table, both `ALTER TABLE table_identifier COMPACT` and `ALTER TABLE table_identifier PARTITION (part_spec) COMPACT` are supported. If the user specifies a partition to compact, we need to keep track of that info in the options. Otherwise, we cannot know which partition needs to be compacted.